keanu-project/src/test/java/io/improbable/keanu/e2e/rocket/RocketTest.java

```diff
@@ -95,4 +77,27 @@ public void shouldYouLaunch() {
         assertEquals(posteriorProbOfAlarm1FalsePositive, 1, 0.001);
     }
 
+    private static class EventCounts {
+        public long oRingFailureCount;
+        public long residualFuelCount;
+        public long alarm1FalsePositiveCount;
+
+        public EventCounts() {
+        }
+
+        public EventCounts(boolean oRingFailure, boolean residualFuel, boolean alarm1FalsePositive) {
+            oRingFailureCount = oRingFailure ? 1 : 0;
+            residualFuelCount = residualFuel ? 1 : 0;
+            alarm1FalsePositiveCount = alarm1FalsePositive ? 1 : 0;
+        }
+
+        public EventCounts combine(EventCounts that) {
+            EventCounts counts = new EventCounts();
+            counts.oRingFailureCount = this.oRingFailureCount + that.oRingFailureCount;
+            counts.residualFuelCount = this.residualFuelCount + that.residualFuelCount;
+            counts.alarm1FalsePositiveCount = this.alarm1FalsePositiveCount + that.alarm1FalsePositiveCount;
+            return counts;
+        }
```

Comment:
I'd just add the values to the current object rather than just creating a new object every time, but I'm pretty old fashioned :)